lib/data_magic/document_builder.rb

```diff
@@ -90,6 +92,17 @@ def fix_field_type(type, value, key=nil)
         new_value
       end
 
+      def parse_boolean(value)
+        case value
+        when "true"
+          true
+        when "false"
+          false
+        else
+          fail InvalidDictionary, "unexpected value '#{value.inspect}' for type '#{type.inspect}' for field '#{key}'"
```

Comment:
It's fine to add this here as a fallback, but the primary place to do this type-checking is in `DataMagic::ErrorChecker.report_wrong_field_type()` . I recommend expanding the code there to handle booleans.